src/pytesmo/validation_framework/adapters.py

```diff
@@ -89,28 +134,43 @@ class MaskingAdapter(BasicAdapter):
     ----------
     cls: object
         has to have a read_ts or read method
-    operator: string
-        one of '<', '<=', '==', '>=', '>', '!='
+    op: str or Callable
+        one of '<', '<=', '==', '>=', '>', '!=' or a function that takes
+        data and threshold as arguments.
     threshold:
         value to use as the threshold combined with the operator
-    column_name: string, optional
+    column_name: str, optional
         name of the column to cut the read masking dataset to
+    kwargs:
+        Additional kwargs are passed to BasicAdapter.
     """
 
-    def __init__(self, cls, op, threshold, column_name = None):
-        super(MaskingAdapter, self).__init__(cls)
+    def __init__(self, cls, op, threshold, column_name=None, **kwargs):
+        super(MaskingAdapter, self).__init__(cls, **kwargs)
+
+        self.op_lookup = {
+            "<": operator.lt,
+            "<=": operator.le,
+            "==": operator.eq,
+            "!=": operator.ne,
+            ">=": operator.ge,
+            ">": operator.gt,
+        }
```

Comment:
Not directly related to your changes, but why is this an attribute? Since this is also used in other adapters, I think it's better to just make this a global variable within this module, maybe something like `_op_lookup`.